astropy/coordinates/matching.py

```diff
@@ -213,7 +213,8 @@ def search_around_3d(coords1, coords2, distlimit, storekdtree="kdtree_3d"):
         The second set of coordinates, which will be searched for matches from
         ``coords1`` within ``seplimit``.  Cannot be a scalar coordinate.
     distlimit : `~astropy.units.Quantity` ['length']
-        The physical radius to search within.
+        The physical radius to search within. It should be a single radius or have the
+        same shape than ``coords1``.
```

Comment:
What is it that prevents broadcasting `distlimit`?

Reply:
We could do it, but it affects the performance, see https://github.com/astropy/astropy/pull/17824#discussion_r1972154489

Reply:
Suppose for the sake of example that `coords1` has the shape `(3, 3, 2)` and the `distlimit` I'd like to use has the shape `(3, 1, 2)`. The docstring says I can't use that `distlimit`, so I am forced to broadcast it myself. I don't understand how that could have a smaller performance penalty than if the function did the broadcasting automatically.

Reply:
In the comment linked, the idea was not to let the improvement implemented here be prevented by the fact that the matching code (in `main`) can basically not deal with multi-dimensional coordinate arrays. Since that needs more work anyway, it seemed reasonable to focus here on just allowing what was possible relatively simply.

But it remains a bit sad that this was not realized when the original matching code was written...